server/src/main/java/org/elasticsearch/Build.java

```diff
@@ -249,6 +249,16 @@ public String qualifiedVersion() {
         return version;
     }
 
+    /**
+     * Remove certain version suffixes. (Just "-SNAPSHOT" at the moment)
+     */
+    public String unqualifiedVersion() {
```

Comment:
Nit: I'd be more comfortable having it as a helper function in InstallPluginAction, but up to you :)

If we want to go this way though, I will consider revisiting Build: `version` in on-prem is currently built by adding -SNAPSHOT, I find it strange to add it and then remove it. Perhaps we could save both `qualifiedVersion` and `unqualifiedVersion` fields in Build, and get rid of `version` at all?

Again, I think it is strange from serverless perspective to have these: personally, I would go with `version` only and adjust where needed (removing `qualifiedVersion()`, which is "strange" at this point) and possibly remove the "special cases" later, if we can. 
On the other hand we do not want to litter code with the same substring stuff over and over. So, up to you!

Reply:
I think you're right about not cluttering up Build. If we find ourselves rewriting these code snippets a lot, we can add a SemVarUtils class to our common lib, but I don't think we need that yet.